plugin/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -2114,9 +2115,9 @@ public void autopersist(@NonNull FlowNode n) throws IOException {
                                 if (owner != null) {
                                     FlowExecution exec = owner.getOrNull();
                                     if (exec instanceof CpsFlowExecution) {
-                                        Set<String> calls = ((CpsFlowExecution) exec).getInternalCalls();
+                                        List<String> calls = ((CpsFlowExecution) exec).getInternalCalls().stream().filter(observed::add).collect(Collectors.toList());
                                         if (!calls.isEmpty()) {
-                                            pw.println("Internal calls for " + run + ":");
+                                            pw.println("Internal calls for " + run + " not previously noted:");
```

Comment:
Have you found the data here to be useful when analyzing support bundles only in general, or was it actually the per-job data that was helpful? 

If it is mainly useful in general I think it would be clearer to just drop the per-job listing entirely and just show all unique internal calls from all jobs. The per-job data seems like it would be too inconsistent to be useful after this change.

If the per-job data is useful, I would consider printing all of the internal calls first with an associated ID number, then printing each job with the ID numbers of associated internal calls, or something like that, so that we can track per-job calls precisely if necessary while still reducing the size of the file.

Reply:
I do not think we need an exhaustive list of jobs or anything. In a bundle I looked at, the false positives were aggravated by being reported from thousands of jobs, resulting in a 24Mib file. But if there is no clue where a call is being used, that would also be a problem.

:thinking: what about listing all calls and then just showing one example caller?

```
staticMethod jenkins.model.Jenkins get (somefolder/somejob #123)
```

That would be enough to help you track down unwanted calls, while keeping the file compact.

Reply:
Sure, that seems fine. I think it would be nice to add a further note that other builds are not listed e.g. `staticMethod jenkins.model.Jenkins get (somefolder/somejob #123 and 4 other builds)` to make it clear that information has been omitted and to give an idea of how widespread the usage is.

Reply:
> and 4 other builds

Thought of that, but this would perhaps be more misleading that helpful because we are only even inspecting the last build of each job.